edx_solutions_api_integration/management/commands/send_progress_leaderboard_notifications.py

```diff
@@ -0,0 +1,105 @@
+"""
+Management command to check users progress and sends notifications
+./manage.py lms send_progress_leaderboard_notifications  --settings=aws --time-range=30
+"""
+
+import logging
+import sys
+from optparse import make_option
+
+from completion_aggregator.models import Aggregator
+from django.conf import settings
+from django.core.management.base import BaseCommand
+from django.utils import timezone
+from edx_notifications.data import NotificationMessage
+from edx_notifications.lib.publisher import (
+    publish_notification_to_user,
+    get_notification_type
+)
+
+from ...models import LeaderBoard
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Command to check users progress and sends notifications
+    """
+    help = 'Check users progress and sends notifications'
+    batch_size = 100
+
+    option_list = BaseCommand.option_list + (
```

Comment:
Please use add_arguments method since `BaseCommand.option_list` is deprecated (https://docs.djangoproject.com/en/1.8/howto/custom-management-commands/#django.core.management.BaseCommand.option_list)